app/models/display_ad.rb

```diff
@@ -37,14 +37,33 @@ class DisplayAd < ApplicationRecord
                              search: "%#{term}%"
                      }
 
-  def self.for_display(area:, user_signed_in:, organization_id: nil, article_tags: [])
-    DisplayAds::FilteredAdsQuery.call(
+  def self.for_display(area:, user_signed_in:, organization_id: nil, article_tags: [], permit_adjacent_sponsors: true)
+    ads_for_display = DisplayAds::FilteredAdsQuery.call(
       display_ads: self,
       area: area,
       organization_id: organization_id,
       user_signed_in: user_signed_in,
       article_tags: article_tags,
+      permit_adjacent_sponsors: permit_adjacent_sponsors,
     )
+
+    # Business Logic Context:
+    # We are always showing more of the good stuff — but we are also always testing the system to give any a chance to
+    # rise to the top. 1 out of every 8 times we show an ad (12.5%), it is totally random. This gives "not yet
+    # evaluated" stuff a chance to get some engagement and start showing up more. If it doesn't get engagement, it
+    # stays in this area.
+
+    # Ads that get engagement have a higher "success rate", and among this category, we sample from the top 15 that
+    # meet that criteria. Within those 15 top "success rates" likely to be clicked, there is a weighting towards the
+    # top ranked outcome as well, and a steady decline over the next 15 — that's because it's not "Here are the top 15
+    # pick one randomly", it is actually "Let's cut off the query at a random limit between 1 and 15 and sample from
+    # that". So basically the "limit" logic will result in 15 sets, and then we sample randomly from there. The
+    # "first ranked" ad will show up in all 15 sets, where as 15 will only show in 1 of the 15.
+    if rand(8) == 1
```

Comment:
`cc` @Ridhwana @lboogie04 I'll explain a bit more below, but I wound up refactoring `filtered_ads_query_spec` quite a bit, and moving this randomization routine into the class here allowed those tests to be a _lot_ more explicit about what was being filtered, without worrying about false-(positive or negative) flaky tests that were passing one minute and failing the next due only to the randomness. 

We _might_ want to consider whether we add more tests to `display_ad_spec`, but testing this randomness is a bit brittle, at best.